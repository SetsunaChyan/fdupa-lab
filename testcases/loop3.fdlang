function main(){
x = 1;

while (l1 < 114) {
    l2 = 0;
    while (l2 < 51) {
        l3 = 0;
        while (l3 < 4) {
            l4 = 0;
            while (l4 < 19) {
                x = x + x;
                l4 = l4 + 1;
                if (x >= 101) {
                    x = x - 101;
                } else {
                    nop;
                }
            }
            l3 = l3 + 1;
        }
        l2 = l2 + 1;
    }
    l1 = l1 + 1;
}

check_interval(x, 79, 79);
check_interval(l1, 114, 114);
check_interval(l2, 51, 51);
check_interval(l3, 4, 4);
check_interval(l4, 19, 19);

check_interval(x, 83, 100);
check_interval(l1, 113, 113);
check_interval(l2, 50, 50);
check_interval(l3, 3, 3);
check_interval(l4, 18, 18);
}
