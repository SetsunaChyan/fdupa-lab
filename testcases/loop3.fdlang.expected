Line 26: YES; x in [79, 79]
Line 27: YES; l1 in [114, 114]
Line 28: YES; l2 in [51, 51]
Line 29: YES; l3 in [4, 4]
Line 30: YES; l4 in [19, 19]
Line 32:  NO; x in [79, 79]
Line 33:  NO; l1 in [114, 114]
Line 34:  NO; l2 in [51, 51]
Line 35:  NO; l3 in [4, 4]
Line 36:  NO; l4 in [19, 19]
